File: DotNET/Endpoint Examples/JSON Payload/decrypted-pdf.cs
/*
 * What this sample does:
 * - Called from Program.cs to remove an open password from a PDF via JSON flow.
 *
 * Setup (environment):
 * - Copy .env.example to .env
 * - Set PDFREST_API_KEY=your_api_key_here
 * - Optional: set PDFREST_URL to override the API region. For EU/GDPR compliance and proximity, use:
 *     PDFREST_URL=https://eu-api.pdfrest.com
 *   For more information visit https://pdfrest.com/pricing#how-do-eu-gdpr-api-calls-work
 *
 * Usage:
 *   dotnet run -- decrypted-pdf /path/to/input.pdf
 *
 * Output:
 * - Prints JSON responses; non-2xx results exit non-zero.
 */
using Newtonsoft.Json.Linq;
using System.Text;

namespace Samples.EndpointExamples.JsonPayload
{
    public static class DecryptedPdf
    {
        public static async Task Execute(string[] args)
        {
            if (args == null || args.Length < 1)
            {
                Console.Error.WriteLine("decrypted-pdf requires <inputFile>");
                Environment.Exit(1);
                return;
            }

            var inputPath = args[0];
            if (!File.Exists(inputPath))
            {
                Console.Error.WriteLine($"File not found: {inputPath}");
                Environment.Exit(1);
                return;
            }

            var apiKey = Environment.GetEnvironmentVariable("PDFREST_API_KEY");
            if (string.IsNullOrWhiteSpace(apiKey))
            {
                Console.Error.WriteLine("Missing required environment variable: PDFREST_API_KEY");
                Environment.Exit(1);
                return;
            }

            var baseUrl = Environment.GetEnvironmentVariable("PDFREST_URL") ?? "https://api.pdfrest.com";
            // Optional immediate deletion of sensitive files
            // Default: false; override with PDFREST_DELETE_SENSITIVE_FILES=true
            var deleteSensitiveFiles = string.Equals(
                Environment.GetEnvironmentVariable("PDFREST_DELETE_SENSITIVE_FILES"),
                "true",
                StringComparison.OrdinalIgnoreCase);

            using (var httpClient = new HttpClient { BaseAddress = new Uri(baseUrl) })
            {
                using (var uploadRequest = new HttpRequestMessage(HttpMethod.Post, "upload"))
                {
                    uploadRequest.Headers.TryAddWithoutValidation("Api-Key", apiKey);
                    uploadRequest.Headers.Accept.Add(new("application/json"));

                    var uploadByteArray = File.ReadAllBytes(inputPath);
                    var uploadByteAryContent = new ByteArrayContent(uploadByteArray);
                    uploadByteAryContent.Headers.TryAddWithoutValidation("Content-Type", "application/octet-stream");
                    uploadByteAryContent.Headers.TryAddWithoutValidation("Content-Filename", Path.GetFileName(inputPath));

                    uploadRequest.Content = uploadByteAryContent;
                    var uploadResponse = await httpClient.SendAsync(uploadRequest);
                    var uploadResult = await uploadResponse.Content.ReadAsStringAsync();

                    Console.WriteLine("Upload response received.");
                    Console.WriteLine(uploadResult);

                    var uploadedID = JObject.Parse(uploadResult)["files"]![0]!["id"]!;

                    using (var decryptRequest = new HttpRequestMessage(HttpMethod.Post, "decrypted-pdf"))
                    {
                        decryptRequest.Headers.TryAddWithoutValidation("Api-Key", apiKey);
                        decryptRequest.Headers.Accept.Add(new("application/json"));
                        decryptRequest.Headers.TryAddWithoutValidation("Content-Type", "application/json");

                        var parameterJson = new JObject
                        {
                            ["id"] = uploadedID,
                            ["current_open_password"] = "password"
                        };

                        decryptRequest.Content = new StringContent(parameterJson.ToString(), Encoding.UTF8, "application/json");
                        var decryptResponse = await httpClient.SendAsync(decryptRequest);
                        var decryptResult = await decryptResponse.Content.ReadAsStringAsync();

                        Console.WriteLine("Processing response received.");
                        Console.WriteLine(decryptResult);

                        // All files uploaded or generated are automatically deleted based on the
                        // File Retention Period as shown on https://pdfrest.com/pricing.
                        // For immediate deletion of files, particularly when sensitive data
                        // is involved, an explicit delete call can be made to the API.
                        //
                        // The following code is an optional step to delete sensitive files
                        // (unredacted, unencrypted, unrestricted, or unwatermarked) from pdfRest servers.
                        if (deleteSensitiveFiles)
                        {
                            using (var deleteRequest = new HttpRequestMessage(HttpMethod.Post, "delete"))
                            {
                                deleteRequest.Headers.TryAddWithoutValidation("Api-Key", apiKey);
                                deleteRequest.Headers.Accept.Add(new("application/json"));
                                deleteRequest.Headers.TryAddWithoutValidation("Content-Type", "application/json");

                                var outId = JObject.Parse(decryptResult)["outputId"]!.ToString();
                                var deleteJson = new JObject { ["ids"] = outId };
                                deleteRequest.Content = new StringContent(deleteJson.ToString(), Encoding.UTF8, "application/json");
                                var deleteResponse = await httpClient.SendAsync(deleteRequest);
                                var deleteResult = await deleteResponse.Content.ReadAsStringAsync();
                                Console.WriteLine(deleteResult);
                            }
                        }
                    }
                }
            }
        }
    }
}

File: Python/Endpoint Examples/JSON Payload/decrypted-pdf.py
import requests
import json

with open('/path/to/file', 'rb') as f:
    upload_data = f.read()

print("Uploading file...")
upload_response = requests.post(url='https://api.pdfrest.com/upload',
                    data=upload_data,
                    headers={'Content-Type': 'application/octet-stream', 'Content-Filename': 'file.pdf', "API-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"})

print("Upload response status code: " + str(upload_response.status_code))

if upload_response.ok:
    upload_response_json = upload_response.json()
    print(json.dumps(upload_response_json, indent = 2))


    uploaded_id = upload_response_json['files'][0]['id']
    decrypt_data = { "id" : uploaded_id, 'current_open_password': 'password' }
    print(json.dumps(decrypt_data, indent = 2))


    print("Processing file...")
    decrypt_response = requests.post(url='https://api.pdfrest.com/decrypted-pdf',
                        data=json.dumps(decrypt_data),
                        headers={'Content-Type': 'application/json', "API-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"})



    print("Processing response status code: " + str(decrypt_response.status_code))
    if decrypt_response.ok:
        decrypt_response_json = decrypt_response.json()
        print(json.dumps(decrypt_response_json, indent = 2))

        # All files uploaded or generated are automatically deleted based on the 
        # File Retention Period as shown on https://pdfrest.com/pricing. 
        # For immediate deletion of files, particularly when sensitive data 
        # is involved, an explicit delete call can be made to the API.
        #
        # The following code is an optional step to delete sensitive files
        # (unredacted, unencrypted, unrestricted, or unwatermarked) from pdfRest servers.

        result_id = decrypt_response_json['outputId']
        delete_data = { "ids": result_id }
        delete_response = requests.post(url='https://api.pdfrest.com/delete',
                            data=json.dumps(delete_data),
                            headers={'Content-Type': 'application/json', "API-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"})
        print(json.dumps(delete_response.json(), indent = 2))

    else:
        print(decrypt_response.text)
else:
    print(upload_response.text)
